### PR TITLE
Sync moveit1 master

### PR DESCRIPTION
**Pull request by henningkayser**
This is a sync following the instructions discussed in https://github.com/ros-planning/moveit2/issues/45. Locally, this now compiles without any issues, currently some tests in `moveit_core` and `ompl` are failing. Also, I commented all newly added tests `collision_detection_bullet`, I'm working on enabling them the next days.

### CODE REVIEW

**Comment by henningkayser** [comment]
I think this is finally ready now. I had to merge master back into this branch because of PRs that were merged after I made the sync. Rebase is not an option (we want to keep history) and I don't think it's worth the effort to repeat the sync on the current master head. A nice thing about this sync is that now a lot more cmake warnings are checked in CI and clang is also enabled. I didn't enable clang-tidy yet because there are some rules that seem to not play nicely with some rclcpp function types (i.e. callback functions using all non-const references) and there are still some fixes to be made caused by the ROS2 migration. I'll handle those in a follow-up.

**Comment by henningkayser** [comment]
Runtime depends on:
* https://github.com/ros-planning/geometric_shapes/pull/147
* https://github.com/ros-planning/moveit_msgs/pull/81

Still looking into more

File: .travis.yml [hunk 1]
```diff
@@ -8,32 +8,29 @@ cache: ccache
...
-      # env: ROS_REPO=ros
+jobs:
+  fast_finish: true
+  include:
+    - env: TEST="clang-format"  # TODO(henningkayser): Enable ament_lint, clang-tidy
+    - env: TEST=code-coverage
+    - env: ROS_DISTRO=eloquent
+    - compiler: clang  # test_ikfast_plugins takes ~10 minutes: include here to keep the main jobs shorter
+      env: CXXFLAGS="-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wno-overloaded-virtual"
+           # TODO(mlautman): Enable once franka_ikfast_plugin and panda_ikfast_plugin are available
```
Comment (davetcoleman) [inline .travis.yml:30]: Unrelated to this PR, but @mlautman can this test be re-enabled now?

File: moveit_core/CMakeLists.txt [hunk 2]
```diff
@@ -2,38 +2,62 @@ cmake_minimum_required(VERSION 3.10.2)
...
+  set(EXTRA_BOOST_COMPONENTS zlib)
+endif()
+find_package(Boost REQUIRED system filesystem date_time thread iostreams ${EXTRA_BOOST_COMPONENTS})
 
 find_package(eigen3_cmake_module REQUIRED)
 find_package(Eigen3 REQUIRED)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")
+
+# TODO(henningkayser): enable bullet
```
Comment (davetcoleman) [inline moveit_core/CMakeLists.txt:39]: can you add more detail to this comment please?
```suggestion
# TODO(henningkayser): enable bullet for MoveIt 2 once X is fixed
```

File: moveit_core/collision_detection_fcl/CMakeLists.txt [hunk 3]
```diff
@@ -50,23 +50,23 @@ if(BUILD_TESTING)
     set(append_library_dirs "${CMAKE_CURRENT_BINARY_DIR};${CMAKE_CURRENT_BINARY_DIR}/../collision_detection;${CMAKE_CURRENT_BINARY_DIR}/../robot_state;${CMAKE_CURRENT_BINARY_DIR}/../robot_model;${CMAKE_CURRENT_BINARY_DIR}/../utils")
   endif()
 
-  ament_add_gtest(test_fcl_collision_detection test/test_fcl_collision_detection.cpp
-    APPEND_LIBRARY_DIRS "${append_library_dirs}")
+  # ament_add_gtest(test_fcl_collision_detection test/test_fcl_collision_detection.cpp
+  #   APPEND_LIBRARY_DIRS "${append_library_dirs}")
```
Comment (davetcoleman) [inline moveit_core/collision_detection_fcl/CMakeLists.txt:54]: Why are you disabling FCL tests??

Comment (henningkayser) [reply moveit_core/collision_detection_fcl/CMakeLists.txt:54]: You almost made me doubt myself ;) I stumbled over this when resolving conflicts and meant to remove it. The test was renamed some time ago https://github.com/ros-planning/moveit/pull/1856/files#diff-8c0f112273f9ce6b5fbf63828202c055R30 and is still enabled a couple lines below. The commented part was the attempted fixup during the alpha migration and can safely be removed. No test coverage lost.

File: moveit_core/CMakeLists.txt (hunk 2 above)
Comment (henningkayser) [reply moveit_core/CMakeLists.txt]: done

File: moveit_core/collision_detection_fcl/CMakeLists.txt (hunk 3 above)
Comment (henningkayser) [reply moveit_core/collision_detection_fcl/CMakeLists.txt]: removed the comment

File: .travis.yml (hunk 1 above)
Comment (henningkayser) [reply .travis.yml:30]: They should be enabled with this https://github.com/ros-planning/moveit2/pull/205

**Comment by henningkayser** [comment]
> This is a really big PR!

Well, apparently a lot has been merged into MoveIt master again ;)
I think we should try to make this more often in smaller chunks so that it's easier to review.

> I skimmed the files. My biggest concern is the disabling of hard-won test coverage

No tests are being removed. Actually, this sync has enabled a bunch of new tests that haven't been run in MoveIt 2 yet.

File: moveit_core/collision_distance_field/src/collision_distance_field_types.cpp [hunk 4]
```diff
@@ -46,8 +46,8 @@ namespace collision_detection
 {
 static const rclcpp::Logger LOGGER =
     rclcpp::get_logger("moveit_collision_distance_field.collision_distance_field_types");
-const rclcpp::Logger PosedBodyPointDecompositionVector::LOGGER = LOGGER;
-const rclcpp::Logger PosedBodySphereDecompositionVector::LOGGER = LOGGER;
+const rclcpp::Logger PosedBodyPointDecompositionVector::logger_ = LOGGER;
```
[the remark this answers is not in the document]
Comment (henningkayser) [reply moveit_core/collision_distance_field/src/collision_distance_field_types.cpp:49]: good point!

**Comment by davetcoleman** [comment]
Great job!

> I think we should try to make this more often in smaller chunks so that it's easier to review.

Perhaps a recurring monthly calendar reminder @henningkayser would make this happen more often?

**Pull request merged**